### PR TITLE
Update Build and Release workflow

### PR DESCRIPTION
**Pull request by WangEdward**
Update Build and Release workflow for Windows, Linux and MacOS

---

**Reference**: release workflow from [https://github.com/modmopet/modmopet](https://raw.githubusercontent.com/modmopet/modmopet/main/.github/workflows/release.yml) (Thank you original author!)

**Known issue**: 

- workflow artifacts will need to be extracted twice. The first archive created by workflow is required to preserve the execution permission. The artifacts for workflow upload process will automatically create another archive outside it during the upload process. This won't cause any problem for release workflow.
- Windows build have chance to fail [fail attempt](https://github.com/WangEdward/PastPaperMaster/actions/runs/5994228989/job/16255793858) | [success attempt](https://github.com/WangEdward/PastPaperMaster/actions/runs/5993969494/job/16254985617), manually rerun the job if it fails.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,24 +1,93 @@
...
-        run: |
-          tar -a -c -f "win-PastPaperMaster-b47.zip" "build/windows/runner/Release"
-          gh release create -t="Draft release build 47" -d -F=".github/docs/release-notes.md" "Build 47" "win-PastPaperMaster-b47.zip"
-        
\ No newline at end of file
+    - name: Checkout
+      uses: actions/checkout@v3
+      
+    - name: set version number (unix)
+      run: echo "VERSION_NUM=$(echo ${{ github.ref }} | cut -d / -f 3)" >> $GITHUB_ENV
```
Comment (micfong-z) [inline .github/workflows/release.yml:36]: Incredible work! I'm not quite sure how this version number works - is it simply a "commit number" count?

**Review by micfong-z — DISMISSED** [review]
Since GitHub currently does not offer Apple Silicon macOS runners, I'll build macOS Universal apps on my local machine for now. Maybe remove all macOS-related workflow?

File: .github/workflows/release.yml (hunk 1 above)
Comment (WangEdward) [reply .github/workflows/release.yml:36]: `github.ref`
> for tags it is refs/tags/<tag_name> ( https://docs.github.com/en/actions/learn-github-actions/contexts#github-context )

`cut -d / -f 3` 
> The cut command is used to extract sections from each line of a file or from piped input. The -d option specifies the delimiter character that separates fields in the input. The -f option specifies which field(s) to extract. The number after -f specifies the field number(s) to extract. In this case, -f 3 specifies that the third field should be extracted. [The -d / option specifies that / is the delimiter character that separates fields in the input](https://www.howtogeek.com/775824/how-to-use-the-linux-cut-command/)

check this out https://github.com/WangEdward/PastPaperMaster/releases/tag/v0.1.51

Comment (micfong-z) [reply .github/workflows/release.yml:36]: Oh, so it's a tag name. I see. Thanks for that!

**Comment by WangEdward** [comment]
> Since gitHub currently does not offer Apple Silicon macOS runners, I'll build macOS Universal apps on my local machine for now. Maybe remove all macOS-related workflow?

The app build by the workflow on intel machine also seems to be universal, you can verify it yourself. Anyway, you can always change the release artifacts manually

**Review by micfong-z — APPROVED** [review]
LGTM

**Pull request merged**